ruma-signatures/src/functions.rs

```diff
@@ -731,13 +744,75 @@ pub fn redact(
     Ok(event)
 }
 
+/// Extracts the server names to check signatures for given event. It will return the sender's
+/// server (unless it's a third party invite) and the event id server (on v1 and v2 room versions)
+fn servers_to_check_signatures(
+    object: &CanonicalJsonObject,
+    version: &RoomVersionId,
+) -> Result<BTreeSet<ServerNameBox>, Error> {
```

Comment:
Changed `HashSet<String>` to `BTreeSet<ServerNameBox>` on https://github.com/ruma/ruma/pull/394/commits/2b15b0218dd8a6ff9aba66a685a3c9dafb46a712